tests/consumers/acero_consumer.py

```diff
@@ -0,0 +1,52 @@
+import pyarrow as pa
+from pyarrow.lib import Table
+
+from .consumer import AbstractConsumer
+
+try:
+    import pyarrow.substrait as substrait
+except ImportError:
+    substrait = None
+
+
+class AceroConsumer(AbstractConsumer):
+    """
+    Implementation of the acero substrait consumer Class for testing
+    """
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def run_substrait_query(substrait_query: str) -> type[Table]:
+        """
+        Run the given substrait query and return the result
+
+        Args:
+            substrait_query: A json formatted string representing the
+            substrait query plan
+
+        Returns:
+            A pyarrow table resulting from running the substrait query plan.
+        """
+        buf = pa._substrait._parse_json_plan(substrait_query)
+        reader = substrait.run_query(buf)
+        result = reader.read_all()
+
+        return result
+
+    @staticmethod
+    def write_to_arrow_binary_file(table: type[Table], filepath: str) -> None:
```

Comment:
```suggestion
    def write_to_arrow_binary_file(table: pa.Table, filepath: Path | str) -> None:
```

You'll have to add a `from __future__ import annotations` to the top of this file for the `|` union operator to work, but it's a good idea to allow both `pathlib.Path` objects and also `str` for file locations